Quantum-Control-Applications/NV-Centers/Confocal-Setup/T1.py

```diff
@@ -0,0 +1,114 @@
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qm.qua import *
+import matplotlib.pyplot as plt
+from configuration import *
+
+###################
+# The QUA program #
+###################
+
+t_min = 4  # in clock cycles units (must be >= 4)
+t_max = 250  # in clock cycles units
+dt = 10  # in clock cycles units
+t_vec = np.arange(t_min, t_max + 0.1, dt)  # +0.1 to include t_max in array
+n_avg = 1e6
+start_from_one = True
+
+with program() as T1:
+    counts1 = declare(int)  # saves number of photon counts
+    counts2 = declare(int)  # saves number of photon counts
+    times1 = declare(int, size=100)
+    times2 = declare(int, size=100)
+    counts_1_st = declare_stream()  # stream for counts
+    counts_2_st = declare_stream()  # stream for counts
+    t = declare(int)  # variable to sweep over in time
+    n = declare(int)  # variable to for_loop
+    n_st = declare_stream()  # stream to save iterations
+
+    play("laser_ON", "AOM")
+    wait(100)
+    with for_(n, 0, n < n_avg, n + 1):
+        with for_(t, t_min, t <= t_max, t + dt):
+            if start_from_one:
+                play("pi", "NV")
+            wait(t, "NV")  # variable delay in spin Echo
+
+            align()
+
+            play("laser_ON", "AOM")
+            measure("readout", "SPCM", None, time_tagging.analog(times1, meas_len, counts1))
+            save(counts1, counts_1_st)  # save counts
+            wait(100, "AOM")
+
+            align()
+
+            if start_from_one:
+                play("pi", "NV")
+            wait(t, "NV")  # variable delay in spin Echo
+            play("pi", "NV")  # Pi pulse to qubit to measure second state
+
+            align()
+
+            play("laser_ON", "AOM")
+            measure("readout", "SPCM", None, time_tagging.analog(times2, meas_len, counts2))
+            save(counts2, counts_2_st)  # save counts
+            wait(100, "AOM")
```

Comment:
It seems that you measure T1 for the two states anyway so I don't understand the purpose of start_from_one... Why would the order matter?

Reply:
It's not the same.
In room-temperature NV experiments, the decay is to a mixed state (not to |0>)
So both |0> and |1> decay.
For both cases, we measure both the population at |0> and at |1> for the same reasons Daniel wrote below...
So we can both initialize to |0> or |1> and measure at |0> or |1>!

Now let me complicate things and say that you can also initialize |-1> and measure at |-1>!
But these measurements are more rare

Reply:
Ok I see, thanks I'll need to read about how NVs work !